### DISCUSSION TITLE
add the min_price as well

### DISCUSSION THREAD

**Discussion by putumutukas** [kotope/aio_energy_management#34] [Ideas]
Hello,

Have been using cheapest hours for a while now and made some adjustment according to my needs and now decided i give this HACS integration a try but found one important thing for me is missing- min_price.
* With the old cheapest hours i just changed the ordering of hours to find the most expensive hours to trigger the houshold to switch to batteries from the inverter
* another thing is that if i would like to find out the N most expensive hours to sell back some electricity from the batteries the current integration is lacking few things

As the changes should not be too hard to implement ie just change > to < or ascending descending in sorting it might be nice and easy addition :)

and obviously i have to note that its an excellent thing that Toni has created !

**Comment by kotope** [comment]
Thanks for the feedback :-)

Not sure if I understood correctly, but there's already inverted functionality that you could find the most expensive hours. Setting the configuration 'inversed' to true will find requested amount of the most expensive hours.

What's missing is the min_price indeed. 
There's max_price that will filter hours off if the price is less than set max price, resulting possible of returning less hours than requested. Is this something you would like along with the inverted functionality?
Like finding amount of hours only if the price is greater than e.g. 3.0cent ?

**Reply by putumutukas** [reply]
yea saw the inversion already to get the most expensive hours but now would really appreciate the min_price so that i can set the minimum price when i would be ready to start selling electricity back to the main grid from my batteries/solar.
Like find N amount of hours where price is greater than e.g. 3.0 cent.

Idea is that i will create 2 calendars:
1. Price good enough to sell electricity (nordpool not below selling cost margin)
2. 2-3 most expensive hours with price above 3.0 cent so it is reasonable to sell back to grid even from batteries

**Comment by kotope** [comment]
Sounds reasonable. I will make an enhancement issue of this and implement it soon as it shouldn't be a big task to do :-)
